src/watchful.ts

```diff
@@ -185,3 +187,8 @@ function linkForDashboard(dashboard: cloudwatch.Dashboard) {
   const cfnDashboard = dashboard.node.defaultChild as cloudwatch.CfnDashboard;
   return `https://console.aws.amazon.com/cloudwatch/home?region=${dashboard.stack.region}#dashboards:name=${cfnDashboard.ref}`;
 }
+
+function hasAlarmAction(alarm: cloudwatch.IAlarm): alarm is cloudwatch.IAlarm & { addAlarmAction: (...actions: cloudwatch.IAlarmAction[]) => void } {
```

Comment:
is the `is cloudwatch.IAlarm` on the right-hand side necessary?

Reply:
Do you mean the right half of `cloudwatch.IAlarm & { addAlarmAction: (...actions: cloudwatch.IAlarmAction[]) => void }`? If we go with a type guard, then yes because `{ addAlarmAction: (...actions: cloudwatch.IAlarmAction[]) => void }` is not assignable to the IAlarm type. The type guard in this case is broadening the type to include `addAlarmAction`.

Reply:
I feel like TypeScript is smart enough to combine the existing information that `alarm is cloudwatch.IAlarm` (from the parameter of `addAlarm`) plus `alarm is { addAlarmAction: (...actions: cloudwatch.IAlarmAction[]) => void }` (from return value of `hasAlarmAction`)

Reply:
So you mean
```typescript
function hasAlarmAction(alarm: cloudwatch.IAlarm): alarm is { addAlarmAction: (...actions: cloudwatch.IAlarmAction[]) => void } {
  return 'addAlarmAction' in alarm;
}
```

That results in the following error indicating that I can't omit IAlarm. Did you mean something else?

<img width="982" alt="Screen Shot 2021-08-31 at 3 31 43 PM" src="https://user-images.githubusercontent.com/1909440/131564435-9ee6898f-4914-47c5-955f-f2cf6837c50d.png">

Reply:
So it is necessary, okay! Thanks for being my guinea pig